### PR TITLE
Updated work manager dependency to androidx

### LINKED ISSUE #3881: error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x4000010 (has extras) } in androidx.work.impl.constraints.trackers.NetworkStateTracker
#### Software and hardware versions 
Collect v1.27.0 and older versions

#### Problem description
https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/4d545f6ee802c3b02ed5b47a2116f5d4?time=last-seven-days&sessionId=5ECF7A2900530001141A53B616220073_DNE_0_v2

```
Fatal Exception: java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x4000010 (has extras) } in androidx.work.impl.constraints.trackers.NetworkStateTracker$NetworkStateBroadcastReceiver@267a3db8
       at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:941)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5930)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1405)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1200)
```

```
Caused by java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@738b27e rejected from java.util.concurrent.ScheduledThreadPoolExecutor@24fa93df[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
       at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2011)
       at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:793)
       at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:298)
       at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:503)
       at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:644)
       at androidx.work.impl.background.systemalarm.WorkTimer.startTimer(WorkTimer.java:82)
       at androidx.work.impl.background.systemalarm.DelayMetCommandHandler.onAllConstraintsMet(DelayMetCommandHandler.java:100)
```

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #3881 

#### What has been done to verify that this works as intended?
Not much since it's something I can't reproduce. I just reviewed implemented changes.

#### Why is this the best possible solution? Were any other approaches considered?
I think it's a bug fixed in [version 2.2.0-rc01](https://developer.android.com/jetpack/androidx/releases/work#2.2.0-rc01) I don't know why we kept arch (Pre-AndroidX) dependencies. If I had forgotten about something @lognaturel @seadowg please let me know but I guess we just missed it. 

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It would be good to check auto-send option for example to confirm that there is no regression.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/build.gradle [hunk 1]
```diff
@@ -321,8 +313,6 @@ dependencies {
     testImplementation "org.powermock:powermock-module-junit4:2.0.5"
     testImplementation "org.powermock:powermock-api-mockito2:2.0.5"
 
-    testImplementation "androidx.arch.core:core-testing:2.1.0"
```
Comment (seadowg) [inline collect_app/build.gradle:324]: I'm surprised we're not using this! This is a current dependency https://developer.android.com/jetpack/androidx/releases/arch. Fine to take it out if nothing is breaking I guess!

**Review by seadowg — APPROVED** [review]
All looks good!

**Comment by grzesiek2010** [comment]
This is one of the most common crashes in v1.27 don't you think it should be added to the point release? @seadowg @lognaturel

**Comment by seadowg** [comment]
@grzesiek2010 makes sense to me!

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids 5.1, 7.1 and 10.

Verified cases:
- autosend from settings
- wifi, cellular, wifi or cellular
- servers: Aggregate, Google Sheet and Central
- autosend defined in form
- stressing server with autosend of 1000 forms
- form with defined submission URL

**Comment by mmarciniak90** [comment]
**Tested with success!**
Verified on Androids: 6.0, 8.1, 9.0

I didn't notice any regression.

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**